### PR TITLE
filter doc ID comparison by date range

### PR DESCRIPTION
**Pull request by snopoke**
Related to https://github.com/dimagi/commcare-hq/pull/20998

This will make it easier to do quick comparisons over specific date ranges.

### CODE REVIEW

File: corehq/apps/data_pipeline_audit/management/commands/compare_doc_ids.py [hunk 1]
```diff
@@ -1,29 +1,71 @@
...
 from couchforms.models import doc_types
 
+DATE_FORMAT = "%Y-%m-%d"
+
+
+def valid_date(s):
+    try:
+        return datetime.strptime(s, DATE_FORMAT)
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
```
Comment (pr33thi) [inline corehq/apps/data_pipeline_audit/management/commands/compare_doc_ids.py]: The error message could have a reminder of the correct format

**Review by pr33thi — APPROVED** [review]
I had 1 small comment

File: corehq/apps/data_pipeline_audit/dbacessors.py [hunk 2]
```diff
@@ -130,44 +132,68 @@ def _get_sql_cases_by_doc_type(domain):
...
+        sql_ids.update(list(queryset.values_list('form_id', flat=True)))
     return sql_ids
 
 
-def get_es_case_ids(domain, doc_type):
-    return _get_es_doc_ids(es.CaseES, domain, doc_type)
+def get_es_case_ids(domain, doc_type, startdate, enddate):
+    datefilter = None
+    if startdate or enddate:
+        datefilter = es.cases.modified_range(gte=startdate, lt=enddate)
```
Comment (emord) [inline corehq/apps/data_pipeline_audit/dbacessors.py]: This should be server_modified_range to match sql

Comment (snopoke) [reply corehq/apps/data_pipeline_audit/dbacessors.py]: Thanks: ec3c85d

**Pull request merged**